### PR TITLE
Docker executor

### PR DESCRIPTION
**Pull request by chrisvdg**
Adds docker executor

### CODE REVIEW

**Comment by chrisvdg** [comment]
uhhm, any tips on the failing tests?

**Comment by ahussein** [comment]
I will check the failing tests, but I am wondering why there are no new tests for this PR ?

**Comment by chrisvdg** [comment]
Are there tests for any of the executors?

**Comment by ahussein** [comment]
Not at the moment, no.

**Comment by ahussein** [comment]
I have merged the latest development into this branch. Tests should be green now.

File: JumpScale9/tools/executor/ExecutorDocker.py [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+                return cls(con)
+        raise KeyError("Container with id or name '%s' not found!" % id_or_name)
+
+    @property
+    def logger(self):
+        return self._logger
+
+    def exists(self, path):
+        if path == "/env.sh":
+            raise RuntimeError("SS")
```
Comment (ahussein) [inline JumpScale9/tools/executor/ExecutorDocker.py]: why we are doing this ?

Comment (chrisvdg) [reply JumpScale9/tools/executor/ExecutorDocker.py]: This has also been done for the SSH Executor, so Geert thought it would be needed here too

**Pull request merged**